frontstage/views/passwords.py

```diff
@@ -23,56 +22,35 @@ def forgot_password():
-            'client_secret': app.config['RAS_FRONTSTAGE_CLIENT_SECRET'],
-            'username': email_address,
-            'password': 'password'
-        }
-        headers = {
-            'Accept': 'application/json',
-            'Content-Type': 'application/x-www-form-urlencoded;charset=UTF-8',
-        }
-
-        oauth2_response = requests.post(url=token_url, data=data,
-                                        headers=headers, auth=(app.config['RAS_FRONTSTAGE_CLIENT_ID'],
-                                                               app.config['RAS_FRONTSTAGE_CLIENT_SECRET']))
-        # Check to see that this user account is not locked
-        if oauth2_response.status_code == 401:
-            oauth2Error = json.loads(oauth2_response.text)
-            if 'Unauthorized user credentials' in oauth2Error['detail']:
-                logger.warning('Email address is not registered')
+        if response.status_code == 401:
+            error_json = json.loads(response.text).get('error')
+            error_message = error_json.get('data', {}).get('detail')
+            if 'Unauthorized user credentials' in error_message:
+                logger.info('Requesting password change for unregistered email on OAuth2 server')
                 template_data = {"error": {"type": {"Email address is not registered"}}}
-                return render_template('passwords/forgot-password.html', _theme='default', form=form,
-                                       data=template_data)
-            elif 'User account locked' in oauth2Error['detail']:
-                logger.warning('Trying to reset password when user account is locked on the OAuth2 server')
+                return render_template('passwords/forgot-password.html', _theme='default', form=form, data=template_data)
+            elif 'User account locked' in error_message:
+                logger.warning('Requesting password change for locked email on OAuth2 server')
             else:
-                logger.error('OAuth 2 server generated 401 which is not understood',
-                             oauth2error=oauth2Error.get('detail'))
-
+                logger.warning('Unknown 401 OAuth2 error')
             return render_template('passwords/reset-password.trouble.html', _theme='default',
                                    data={"error": {"type": "failed"}})
-        if oauth2_response.status_code != 201:
-            logger.error('Unknown error from the OAuth2 server')
-            raise ExternalServiceError(oauth2_response)
-
-        url = app.config['RAS_PARTY_RESET_PASSWORD_REQUEST'].format(app.config['RAS_PARTY_SERVICE'])
-        response = requests.post(url, auth=app.config['BASIC_AUTH'], json=post_data, verify=False)
-        if response.status_code == 404:
-            logger.warning('Email address is not registered')
+
+        elif response.status_code == 404:
+            logger.error('Requesting password change for email registered on OAuth2 server but not in party service')
```

Comment:
Should this be error level logging or just info?

Reply:
It's logged at error level because Frontstage API will only return 404 if the user exists in the OAuth server but not in the party service. This should never happen, if it does though something has gone wrong with the user's account and should be investigated.